### DISCUSSION TITLE
How to automate the RunReportJob programmatically

### DISCUSSION THREAD

**Discussion by rakeshSgr** [Sunbird-Ed/Community#560] [Bugs] [answered]
Hi,

As part of the 6.0 release, we are automating the creation of admin reports and program dashboard reports using a Python script. In the script, we make API calls to create backend and frontend configurations.

After running the script, in order to make the reports available, we need to execute the RunReportJob service in Jenkins. This service, accessible at http://10.20.0.14:8080/jenkins/job/Deploy/job/dev/job/DataPipeline/job/RunReportJob/, is responsible for executing the Druid query associated with the reports. It retrieves the query results and converts them into CSV and JSON formats. These output files are then pushed to the cloud storage.

Running the RunReportJob service ensures that the latest data is fetched and made available for generating the reports. Currently, we need to manually run this job for each report.

@manjudr @kumarks1122
When we create a new report, we need to manually run it via Jenkins to create a cron entry.
Please let us know, how we do it if not via Jenkins

CC : @aks30 @vijiurs @adithyadinesh0412 @santhosh-tg @tsprasath

**Comment by chitranshu-keshav** [comment]
@manjudr @kumarks1122 can you please get back on this

**Comment by Krishnaj20** [comment]
@rakeshSgr - Is your query responded? If this belongs to obsrv please raise this query in the obsrv bb forums.

@vinukumar-vs

**Reply by Krishnaj20** [reply, marked as answer]
@rakeshSgr - since we haven't had any response marking this thread as answered. Please feel free to raise a new thread if you have any follow up queries.
